eng/scripts/Collect-Changelogs.ps1

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+#############  Collect Changelog script ###################
+#
+# This script helps to create the Language release notes by collectiong each service's release notes
+# and generating one only release notes with the required format and links to be used to publish.
+#
+# Usage:
+#  - Run from powershell terminal: `pwsh ./Collect-Changelogs.ps1`
+#  - Enter the Month as a number (i.e. use 11 for November)
+#
+# That's it, the script will use the current year and print out the Release Notes in the standard output.
+#
+###########################################################
+
+[CmdletBinding()]
+param(
+  [Parameter(Mandatory=$true)]
+  [ValidateRange(1, 12)]
+  [int] $Month
+)
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../..";
+. ${repoRoot}\eng\common\scripts\SemVer.ps1
+. ${repoRoot}\eng\common\scripts\ChangeLog-Operations.ps1
+$InstallNotes = "";
+$ReleaseNotes = "";
+
+$date = Get-Date -Month $month -Format "yyyy-MM"
+$date += "-\d\d"
+
+Get-ChildItem "$repoRoot/sdk" -Filter CHANGELOG.md -Recurse | Sort-Object -Property Name | % {
+    
+    $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $_ 
+    $package = $_.Directory.Name
+    $serviceDirectory = $_.Directory.Parent.Name
+
+    foreach ($changeLogEntry in $changeLogEntries.Values)
+    {
+        if ($changeLogEntry.ReleaseStatus -notmatch $date)
+        {
+            
+            continue;
+        }
+
+        $version = $changeLogEntry.ReleaseVersion
+        $githubAnchor = $changeLogEntry.ReleaseTitle.Replace("## ", "").Replace(".", "").Replace("(", "").Replace(")", "").Replace(" ", "-")
+
+        $ReleaseNotes += "### $package [Changelog](https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/$serviceDirectory/$package/CHANGELOG.md#$githubAnchor)`n"
```

Comment:
Did you verify this works as expected? How do we test it?

When I run it locally, I get an error:
```text
LogError : ChangeLog[CHANGELOG.md] does not exist
At E:\GitHub\Fork\azure-sdk-for-cpp\eng\common\scripts\ChangeLog-Operations.ps1:18 char:5
```

Reply:
Yes, I even used for creating the last November release notes here: https://azure.github.io/azure-sdk/releases/2020-11/cpp.html#release-highlights

Make sure that the Month number you enter is 9, 10 or 11, since we don't have release notes from before Sep-2020.

Reply:
I must not be doing it correctly then. What directory do I need to be in to run the script?

Can you share the exact steps you took after cloning the C++ repo.

Reply:
```bash
cd eng/scripts
pwsh Collect....
```
That should be enough.  Yes, you have to run it from the same folder where the script is.  I think it takes your current path to look for the release notes from the repo

Reply:
No luck. Can you try running it on Windows?

![image](https://user-images.githubusercontent.com/6527137/102536217-295d0e80-405e-11eb-994e-c625d1ceeb93.png)

Reply:
Well, @ahsonkhan , this is crazy! but it turns out there is this Cross-platform powershell which is different from the one that is installed by default on Windows.
The script requires the this other cross-platform pwsh to run.
You need to go here: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7.1 
Install this powershell and use it to run the script.

This is why I was able to run it from linux, because the pwsh I get there is the one that works :)

Reply:
![image](https://user-images.githubusercontent.com/24213737/102550733-29b3d480-4073-11eb-9129-b4dcd65bfb98.png)

Reply:
Can we write the script in a way that it runs on both? I am guessing it is just some path setting issue (maybe the directory separators).

Reply:
For example, the following works (though the final results has different ordering):
```pwsh
    $directoryPath = $_.Directory.FullName
    $filename = "/CHANGELOG.md"

    $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $($directoryPath + $filename)
```

Reply:
I think there is no benefit on doing such.
The /eng scripts are only to be used internally. 
We don't expect end-users to call/use them.

Usually the scripts are required as part of the CI pipelines.  And in the Azure DevOps nodes, we install the cross-platform powershell all the time for all the current scripts.

I don't think we need to do this.

Keep in mind that we would need to update all the scripts from /eng folder, not only this one.
This script calls at least 2 other scritps from the common folder

Reply:
> Keep in mind that we would need to update all the scripts from /eng folder, not only this one.
This script calls at least 2 other scritps from the common folder

Looks like the fix here is just for the argument passed in to the other script, so pretty isolated, but I get your point.